src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -124,6 +127,9 @@ AttributeSyntax attribute
 					return ( (GenericNameSyntax)typeSynt )
 						.TypeArgumentList.Arguments.First()
 						.WithTriviaFrom( typeSynt );
+				case "TaskCanceledException":
+					m_removeCatchClause = !m_disableTaskRunWarningFlag;

```

Comment:
This has the unfortunate side effect that if anything is wrapped in Task.Run() in the entire method by the generator, then we will not remove a TaskCanceledException catch clause.

I think this is acceptable because it would be pretty rare, and we're doing it in the safer direction (it's fine if we still have a catch even if it's never reached, not fine if we remove a catch that could be reached). It also reduces generator complexity.